### PR TITLE
memory: pylint and code improvements

### PR DESCRIPTION
**Pull request by harish-24**
Patch makes sure memory tests are compliant with pylint and has some code improvements.

Signed-off-by: Harish <harish@linux.vnet.ibm.com>

### CODE REVIEW

File: memory/memhotplug.py [hunk 1]
```diff
@@ -65,19 +62,19 @@ def get_hotpluggable_blocks(path, ratio):
         if memory.is_hot_pluggable(block):
             mem_blocks.append(block)
 
-    def f(num):
+    def percent(num):
```
Comment (narasimhan-v) [inline memory/memhotplug.py]: docstring please

File: memory/dma_memtest.py [hunk 2]
```diff
@@ -49,6 +49,7 @@ def setUp(self):
         copies of the linux kernel that will be uncompressed.
         """
         self.nfail = 0
+        self.base_dir = None
```
Comment (narasimhan-v) [inline memory/dma_memtest.py]: why is this initialised to None ?
It can be `self.base_dir = os.path.join(self.tmpdir, 'linux.orig')` here, right ?

Comment (harish-24) [reply memory/dma_memtest.py]: Right. Will move it after initializing `self.tmpdir`.

**Comment by harish-24** [comment]
@narasimhan-v Pushed with changes, Thanks.

File: memory/memhotplug.py (hunk 1 above)
Comment (narasimhan-v) [reply memory/memhotplug.py]: @harish-24

Comment (harish-24) [reply memory/memhotplug.py]: @narasimhan-v Changes pushed. Forgot to put it here.

Comment (narasimhan-v) [reply memory/memhotplug.py]: @harish-24 this function is doing the reverse of percent.

Comment (harish-24) [reply memory/memhotplug.py]: @narasimhan-v pushed with appropriate definition.

Comment (narasimhan-v) [reply memory/memhotplug.py]: @harish-24 Thanks, I am merging this. Maybe you can send a patch that optimises the function later.

**Pull request merged**